src/test/components/Password.test.tsx

```diff
@@ -0,0 +1,163 @@
+import React from "react";
+import { render, fireEvent } from "@testing-library/react";
+import Password, {
+  PasswordProps,
+} from "../../app/components/elements/inputs/Password";
+
+describe("Password Component", () => {
+  let setIsPasswordVisibleMock: jest.Mock;
+  let onChange: jest.Mock;
+  let onBlur: jest.Mock;
+  let onFocus: jest.Mock;
+  let passwordInput: HTMLInputElement;
+  let toggleIcon: HTMLElement | null;
+
+  beforeEach(() => {
+    setIsPasswordVisibleMock = jest.fn();
+    onChange = jest.fn();
+    onBlur = jest.fn();
+    onFocus = jest.fn();
+  });
+
+  const passwordFieldTypes = ["New", "Confirm", "Current"] as const;
+
+  const renderPassword = (props: Partial<PasswordProps> = {}) => {
+    const defaultProps: PasswordProps = {
+      passwordFieldType: "New",
+      isPasswordVisible: false,
+      setIsPasswordVisible: setIsPasswordVisibleMock,
+      onChange,
+      onBlur,
+      onFocus,
+      ...props,
+    };
+
+    return render(<Password {...defaultProps} />);
+  };
+
+  describe("Event Handling", () => {
+    test("handles onChange, onBlur, and onFocus events when typing", () => {
+      const { container } = renderPassword();
+
+      passwordInput = container.querySelector(
+        'input[name="new-password"]'
+      ) as HTMLInputElement;
+      fireEvent.change(passwordInput, { target: { value: "newPassword123" } });
+      expect(onChange).toHaveBeenCalledTimes(1);
+
+      fireEvent.blur(passwordInput);
+      expect(onBlur).toHaveBeenCalledTimes(1);
+
+      fireEvent.focus(passwordInput);
+      expect(onFocus).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe("Toggle Icon", () => {
+    test("shows toggle icon and handles click, changing password input type", () => {
+      const { getByTestId } = renderPassword({
+        showToggleIcon: true,
+      });
+
+      toggleIcon = getByTestId("show-password-icon");
+      expect(passwordInput).toHaveAttribute("type", "password");
+      fireEvent.click(toggleIcon);
+      expect(setIsPasswordVisibleMock).toHaveBeenCalledTimes(1);
+      expect(setIsPasswordVisibleMock).toHaveBeenCalledWith(true);
+
+      const { container: updatedContainer } = renderPassword({
```

Comment:
This is a bit confusing, because you're essentially rendering a new Password component, both of them share the same `setIspasswordVisibleMock`, and you've done this as part of the same test. It would have been better to do this as a separate test. You can leave it as it is, but please take it into account for future tests.

Reply:
I went ahead and changed this into two separate tests for setting the password as visible / hidden